src/core/Akka/Dispatch/ChannelSchedulerExtension.cs

```diff
@@ -0,0 +1,389 @@
+
+            _controlTask = Task.Factory.StartNew(ControlAsync, _cts.Token,
+                TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning,
+                TaskScheduler.Default).Unwrap();
+        }
+
+        public TaskScheduler GetScheduler(TaskSchedulerPriority priority)
+        {
+            switch (priority)
+            {
+                case TaskSchedulerPriority.Normal:
+                    return _normalScheduler;
+                case TaskSchedulerPriority.Realtime:
+                case TaskSchedulerPriority.High:
+                case TaskSchedulerPriority.AboveNormal:
+                    return _highScheduler;
+                case TaskSchedulerPriority.BelowNormal:
+                case TaskSchedulerPriority.Low:
+                    return _lowScheduler;
+                case TaskSchedulerPriority.Background:
+                //case TaskSchedulerPriority.Idle:
+                    return _idleScheduler;
+                default:
+                    throw new ArgumentException(nameof(priority));
+            }
+        }
+
+        private async Task ControlAsync()
+        {
+            var highReader = _highScheduler.Channel.Reader;
+            var normalReader = _normalScheduler.Channel.Reader;
+            var lowReader = _lowScheduler.Channel.Reader;
+            var idleReader = _idleScheduler.Channel.Reader;
+
+            var readTasks = new Task<bool>[] {
+                highReader.WaitToReadAsync().AsTask(),
+                normalReader.WaitToReadAsync().AsTask(),
+                lowReader.WaitToReadAsync().AsTask(),
+                idleReader.WaitToReadAsync().AsTask()
+            };
+
+            Task<bool> readTask;
+
+            do
+            {
+                //schedule coworkers
+                ScheduleCoWorkers("control");
+
+                //main worker
+                DoWork(0);
+
+                //wait on coworker exit
+                await Task.WhenAll(_coworkers).ConfigureAwait(false);
+
+                //stop timer
+                if (!_cts.IsCancellationRequested)
+                    _timer.Change(Timeout.Infinite, Timeout.Infinite);
+
+                //reset read events
+                if (readTasks[0].IsCompleted)
+                    readTasks[0] = highReader.WaitToReadAsync().AsTask();
+                if (readTasks[1].IsCompleted)
+                    readTasks[1] = normalReader.WaitToReadAsync().AsTask();
+                if (readTasks[2].IsCompleted)
+                    readTasks[2] = lowReader.WaitToReadAsync().AsTask();
+                if (readTasks[3].IsCompleted)
+                    readTasks[3] = idleReader.WaitToReadAsync().AsTask();
+
+                readTask = await Task.WhenAny(readTasks).ConfigureAwait(false);
+            }
+            while (readTask.Result && !_cts.IsCancellationRequested);
+        }
+
+        private void ScheduleCoWorkers(object state)
```

Comment:
Please provide a comment explaining, generally, how this method is supposed to work and what its configurable limitations are.

Not thinking about this for users so much as other contributors / maintainers.

Reply:
Can I then include todo and ideas comments too?

Reply:
Of course!